src/core/lib/event_engine/posix_engine/posix_endpoint.cc

```diff
@@ -122,39 +123,55 @@ ssize_t TcpSend(int fd, const struct msghdr* msg, int* saved_errno,
 
 #define CAP_IS_SUPPORTED(cap) (prctl(PR_CAPBSET_READ, (cap), 0) > 0)
 
+uint64_t ParseUlimitMemLockFromFile(std::string file_name) {
+  std::ifstream limits_file(file_name);
```

Comment:
let's use grpc_load_file here to keep it consistent with everywhere else we read files in production code
(and avoid some bloat!)

Reply:
Done. I created two separate files src/core/lib/gprpp/load_file{.h,.cc} to handle loading data from files. It returns a grpc_core::Slice type.